detector/detector.go

```diff
@@ -265,6 +255,31 @@ func DetectPkgCves(r *models.ScanResult, ovalCnf config.GovalDictConf, gostCnf c
 	return nil
 }
 
+// isPkgCvesDetactable checks whether CVEs is detactable with gost and oval from the result
+func isPkgCvesDetactable(r *models.ScanResult) bool {
+	if reuseScannedCves(r) {
```

Comment:
I think that OVAL and gost scans are skipped when using constant.Raspbian.

Reply:
How about implementing it without `reuseScannedCves`?
```diff
diff --git a/detector/detector.go b/detector/detector.go
index 49a2004b..f7825981 100644
--- a/detector/detector.go
+++ b/detector/detector.go
@@ -257,27 +257,26 @@ func DetectPkgCves(r *models.ScanResult, ovalCnf config.GovalDictConf, gostCnf c
 
 // isPkgCvesDetactable checks whether CVEs is detactable with gost and oval from the result
 func isPkgCvesDetactable(r *models.ScanResult) bool {
-	if reuseScannedCves(r) {
-		logging.Log.Infof("Use CVEs as it as. Skip OVAL and gost detection")
-		return false
-	}
-
-	if r.Family == constant.ServerTypePseudo {
-		logging.Log.Infof("pseudo type. Skip OVAL and gost detection")
-		return false
-	}
-
 	if r.Release == "" {
 		logging.Log.Infof("r.Release is empty. detect as pseudo type. Skip OVAL and gost detection")
 		return false
 	}
-
-	if len(r.Packages)+len(r.SrcPackages) == 0 {
-		logging.Log.Infof("Number of packages is 0. Skip OVAL and gost detection")
+	if r.ScannedBy == "trivy" {
+		logging.Log.Infof("r.ScannedBy is trivy. Skip OVAL and gost detection")
 		return false
 	}
 
-	return true
+	switch r.Family {
+	case constant.FreeBSD, constant.ServerTypePseudo:
+		logging.Log.Infof("%s type. Skip OVAL and gost detection", r.Family)
+		return false
+	default:
+		if len(r.Packages)+len(r.SrcPackages) == 0 {
+			logging.Log.Infof("Number of packages is 0. Skip OVAL and gost detection")
+			return false
+		}
+		return true
+	}
 }
 
 // DetectGitHubCves fetches CVEs from GitHub Security Alerts
```

Reply:
thanks for review.

Do you mean to be better to output the reason for skipping in detail?
And, we need to add a condition to this function when OS is added that reuses the scan results, is it okay?
(I concern the risk to miss to add condition.)

Reply:
"reuseScannedCves" seems to be for skipping the initialization of ScannedCves in environments where CVEs are detected during scanning (e.g. FreeBSD).
I did not think it necessary to use "reuseScannedCves" because it is not what I wanted to implement this time.

Reply:
sry, I understood that reuseScanCves () is unrelated in this function